qiskit/algorithms/state_fidelities/compute_uncompute.py

```diff
@@ -0,0 +1,147 @@
+        """
+        circuit = circuit_1.compose(circuit_2.inverse())
+        circuit.measure_all()
+        return circuit
+
+    def _run(
+        self,
+        circuits_1: Sequence[QuantumCircuit],
+        circuits_2: Sequence[QuantumCircuit],
+        values_1: Sequence[Sequence[float]] | None = None,
+        values_2: Sequence[Sequence[float]] | None = None,
+        **run_options,
+    ) -> StateFidelityResult:
+        r"""
+        Computes the state overlap (fidelity) calculation between two
+        (parametrized) circuits (first and second) for a specific set of parameter
+        values (first and second) following the compute-uncompute method.
+
+        Args:
+            circuits_1: (Parametrized) quantum circuits preparing :math:`|\psi\rangle`.
+            circuits_2: (Parametrized) quantum circuits preparing :math:`|\phi\rangle`.
+            values_1: Numerical parameters to be bound to the first circuits.
+            values_2: Numerical parameters to be bound to the second circuits.
+            run_options: Backend runtime options used for circuit execution. The order
+            of priority is: run_options in ``run`` method > fidelity's default
+            run_options > primitive's default setting.
+            Higher priority setting overrides lower priority setting.
+
+        Returns:
+            The result of the fidelity calculation.
+
+        Raises:
+            ValueError: At least one pair of circuits must be defined.
+            QiskitError: If the sampler job is not completed successfully.
+        """
+
+        circuits = self._construct_circuits(circuits_1, circuits_2)
+        if len(circuits) == 0:
+            raise ValueError(
+                "At least one pair of circuits must be defined to calculate the state overlap."
+            )
+        values = self._construct_value_list(circuits_1, circuits_2, values_1, values_2)
+
+        # The priority of run options is as follows:
+        # run_options in `evaluate` method > fidelity's default run_options >
+        # primitive's default run_options.
+        run_opts = copy(self._default_run_options)
+        run_opts.update(**run_options)
+
+        job = self._sampler.run(circuits=circuits, parameter_values=values, **run_opts)
+
+        result = job.result()
+        status = job.status()
+
+        if status is JobStatus.DONE:
+            raw_fidelities = [prob_dist.get(0, 0) for prob_dist in result.quasi_dists]
+            fidelities = self._truncate_fidelities(raw_fidelities)
+
+            return StateFidelityResult(
+                fidelities=fidelities,
+                raw_fidelities=raw_fidelities,
+                metadata=result.metadata,
+                run_options=run_opts,
+            )
+        else:
+            raise QiskitError(
+                f"The sampler fidelity job was not completed succesfully. "
+                f"Job status = {status}."
+            )
```

Comment:
I am not sure about this way of handling the status. Should we rather be passing status on - e.g this ends up CANCELLED because the underlying job was cancelled. Or maybe the status return from fidelity job here could be ERROR with a presumably a reason that says underlying job was not successful and was cancelled/had error or whatever.  Ie indicate these via the PrimitiveJob that gets returned from here in the same way the primitives themselves report back in a job since this has a consistent interface in regards jobs etc. The _run/run interface might have to adapt to cater to this of course.